### PR TITLE
(BREAKING CHANGE) RTC 536910 - [source] Remove ubuntuAptSource INBM configuration tag and underlying code; replaced with source command

### PR DESCRIPTION
**Pull request by tsirlapu**
RTC 536910

Remove the UbuntuAptSource XML tag from the intel_manageability.conf file and its associated tag with it

_The PR review is to check for sustainability and correctness.  Sustainability is actually more business critical as correctness is largely tested into the code over time.   Its useful to keep in mind that SW often outlives the HW it was written for and engineers move from job to job so it is critical that code developed for Intel be supportable across many years.  It is up to the submitter and reviewer to look at the code from a perspective of what if we have to debug this 3 years from now after the author is no longer available and defect databases have been lost.  Yes, that happens all the time when we are working with time scales of more than 2 years.  When reviewing your code it is important to look at it from this perspective._

Author Mandatory (to be filled by PR Author/Submitter)
------------------------------------------------------
- Developer who submits the Pull Request for merge is required to mark the checklist below as applicable for the PR changes submitted.  
- Those checklist items which are not marked are considered as not applicable for the PR change.  
- Items marked with an asterisk suffix are mandatory items to check and if not marked will be treated as non-compliant pull requests by the developers for Inner Source Development Model (ISDM) compliance

### PULL DESCRIPTION
_Provide a 1-2 line brief overview of the changes submitted through the Pull Request..._

### REFERENCES
Reference URL for issue tracking (JIRA/HSD/Github): **\<URL to be filled>**
- [ ] **_Added label to the Pull Request following the template: ISDM\_\<Complexity>\*_** \
	Note-1: Depending on complexity of code changes, use the suitable word for complexity: Low/Medium/High \
	Example: PR for Slim boot loader project with medium complexity can have the label as: ISDM_Medium	
- [ ] Added label to the Pull Request for easier discoverability and search
- [ ] RTC or HSD number will be included in final merge. HSD must always be included if available.
- [ ] Changelogs are updated (or N/A if not customer visible)
- [ ] inbm/log_changes.txt and inbm-vision/log_changes.txt are updated for potentially Validation-breaking log changes (or N/A if none)

### CODE MAINTAINABILITY
- [ ] **_Commit Message meets guidelines as indicated in the URL\*_**
	- (https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md)
- [ ] **_Every commit is a single defect fix and does not mix feature addition or changes\*_**
- [ ] Added required new tests relevant to the changes
	- [ ] PR contains URL links to functional tests executed with the new tests 
- [ ] Updated Documentation as relevant to the changes
- [ ] Updated Build steps/commands changes as relevant
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide description)
- [ ] Specific instructions or information for code reviewers (If any):
- [ ] Run 'go fmt' or format-python.sh as applicable.
- [ ] New/modified methods and functions should have type annotations on signatures as applicable
- [ ] New/modified methods must have appropriate doc strings (language dependent)

### APPLICATION SPECIFIC
- [ ] Does PR change default config files under /etc? If so, will application still work after an upgrade that leaves /etc alone, like a Mender upgrade?
- [ ] Is cloud UI changed? If so, are cloud definition files updated?

Maintainer Mandatory (to be filled by PR Reviewer/Approving Maintainer)
-----------------------------------------------------------------------
- Maintainer who approves the Pull Request for merge is required to mark the checklist below as appropriate for the PR change reviewed as key proof of attestation indicating reasons for merge. 
- Those checklist items which are not marked are considered as not applicable for the PR change. 
- Items marked with an asterisk suffix are mandatory items to check and if not marked will be treated as non-compliant pull requests by the maintainers for ISDM compliance.

### QUALITY CHECKS
- [ ] Architectural and Design Fit
- [ ] **_Quality of code (At least one should be checked as applicable)\*_**
	- [ ] Commit Message meets guidelines
	- [ ] PR changes adhere to industry practices and standards
	- [ ] Error and exception code paths implemented correctly
	- [ ] Code reviewed for domain or language specific anti-patterns
	- [ ] Code is adequately commented
	- [ ] Code copyright is correct
	- [ ] Confusing logic is explained in comments
	- [ ] Commit comment can be used to design a new test case for the changes
- [ ] **_Test coverage shows adequate coverage with required CI functional tests pass on all supported platforms\*_**
- [ ] **_Static code scan report shows zero critical issues\*_**
- [ ] Integration tests are passing

### CODE REVIEW IMPACT
- Summary of Defects Detected in Code Review: **\<%P1*xx,P2*xx,P3*xx,P4*xx%>** \
Note P1/P2/P3/P4 denotes severity of defects found (Showstopper/High/Medium/Low) and xx denotes number of defects found

### SECURITY CHECKS
Please check if your PR fulfills the following requirements:

- [ ] Follow best practices when handling primitive data types
- [ ] Configure minimal permissions when opening pipes and ports
- [ ] Check contents within input structures are valid before use
- [ ] All forms of input validated
- [ ] Avoid inter-process race conditions
- [ ] Error and exception handling implemented
- [ ] Defend against Canonical Representation Issues - Any paths utilized?
- [ ] Follow 'secure by default' - Any configuration values added
- [ ] Fail safe - Any failure scenarios?
- [ ] Clean up temporary files - Any temporary files being used?

# _Code must act as a teacher for future developers_

### CODE REVIEW

File: docs/Configuration Parameters.md [hunk 1]
```diff
@@ -49,7 +49,8 @@ The below tables represent the different sections of the configuration file.
 | token                | /etc/opt/csl/csl-node/long-lived-token | path to token      |
 | certFile             |     /etc/ssl/certs/csl-ca-cert.pem     | path the cert file |
 
-### SOTA
+Note: ubuntuAptSource tag is no longer used in/After 4.2.0 Version
```
Comment (nex-maximus) [inline docs/Configuration Parameters.md:52]: The removal note about `ubuntuAptSource` tag is clear, but ensure that the documentation is consistent across all files and that no outdated references remain. [medium]

File: docs/In-Band Manageability Developer Guide.md [hunk 2]
```diff
@@ -534,6 +534,8 @@ To send the whole manifest with edited parameters at once,
 -   To overwrite the existing configuration file with a new one then use
     the **Load Element Manifest**.
 
+Note: ubuntuAptSource tag is no longer used in/After 4.2.0 Version
+
 The following commands are useful to append, remove values for
 parameters that have multiple values. Parameters that have multiple
 values are **trustedRepositories**, **sotaSW** and **ubuntuAptSource**.
```
Comment (nex-maximus) [inline docs/In-Band Manageability Developer Guide.md:541]: Update the documentation to remove the example usage of `ubuntuAptSource` if it's no longer relevant. [medium]

File: docs/Configuration Parameters.md (hunk 1 above)
Comment (gblewis1) [inline docs/Configuration Parameters.md:52]: @tsirlapu this is unclear--which version is the first one that no longer uses the tag?

File: docs/Configuration Parameters.md [hunk 3]
```diff
@@ -49,7 +49,8 @@ The below tables represent the different sections of the configuration file.
 | token                | /etc/opt/csl/csl-node/long-lived-token | path to token      |
 | certFile             |     /etc/ssl/certs/csl-ca-cert.pem     | path the cert file |
 
-### SOTA
+Note: ubuntuAptSource tag is no longer used in/After 4.2.0 Version 
+
 | Key                    |     Default Value      | Description                                                                              |
 |:-----------------------|:----------------------:|:-----------------------------------------------------------------------------------------|
 | ubuntuAptSource        | http://yoururl/ubuntu/ | Location used to update Ubuntu                                                           |
```
Comment (gblewis1) [inline docs/Configuration Parameters.md:56]: remove

File: docs/In-Band Manageability Developer Guide.md (hunk 2 above)
Comment (gblewis1) [inline docs/In-Band Manageability Developer Guide.md:541]: remove ubuntuAptSource

File: docs/In-Band Manageability User Guide - ThingsBoard.md [hunk 4]
```diff
@@ -1170,6 +1170,7 @@ Events Log in the dashboard below the OTA buttons.
 
 **Required Fields**: Command and Path
 
+Note: ubuntuAptSource tag is no longer used in/After 4.2.0 Version
 
 **NOTE:**
 > - Append is only applicable to three configuration tags i.e
```
Comment (gblewis1) [inline docs/In-Band Manageability User Guide - ThingsBoard.md:1176]: remove ubuntuAptSource

File: docs/Manifest Parameters.md [hunk 5]
```diff
@@ -853,6 +853,8 @@ The query command can be used to gather information about the system and the Vis
 
 #### Configuration Append Manifest Example
 
+Note: ubuntuAptSource tag is no longer used in/After 4.2.0 Version
+
 * Append is only applicable to three configuration tags, for example,
```
Comment (gblewis1) [inline docs/Manifest Parameters.md:858]: remove ubuntuAptSource

File: docs/Configuration Parameters.md (hunk 1 above)
Comment (tsirlapu) [reply docs/Configuration Parameters.md]: Fixed

File: docs/Configuration Parameters.md (hunk 3 above)
Comment (tsirlapu) [reply docs/Configuration Parameters.md]: Fixed

File: docs/Configuration Parameters.md (hunk 1 above)
Comment (gblewis1) [reply docs/Configuration Parameters.md]: > Fixed

Text still reads "Note: ubuntuAptSource tag is no longer used in/After 4.2.0 Version". I would suggest just "Note: ubuntuAptSource tag has been removed"

Comment (gblewis1) [reply docs/Configuration Parameters.md]: (also, please change the other instances of "Note: ubuntuAptSource tag is no longer used in/After 4.2.0 Version")

Comment (tsirlapu) [reply docs/Configuration Parameters.md]: @gblewis1 "Note: ubuntuAptSource tag has been removed" will not mention from which version it has been removed, is that okay?

File: inbm/Changelog.md [hunk 6]
```diff
@@ -10,6 +10,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Changed
  - RTC 536078 - Added package list option to inbc, cloud, and internal manifest. This allows SOTA to run an install/upgrade command on a set of individual packages rather than all installed packages.
+ - RTC  536910 - [source] Remove ubuntuAptSource configuration tag and underlying code
```
Comment (gblewis1) [inline inbm/Changelog.md:13]: highlight this is a potentially breaking change and anyone relying on ubuntuAptSource needs to use the new 'source' mechanism

Comment (gblewis1) [reply inbm/Changelog.md]: @tsirlapu i see you marked the conversation as resolved, but I don't see that you're highlighting this as a breaking change

**Comment by gblewis1** [comment]
@tsirlapu I see that you've marked ubuntuAptSource as removed from the 4.2.0 version, but that's actually not true; 4.2.0 still has ubuntuAptSource. It would be better to simply say it's removed as the person reading the docs would be reading them from the new version that has it removed, and not 4.2.0 or earlier.

File: inbm/Changelog.md (hunk 6 above)
Comment (tsirlapu) [reply inbm/Changelog.md]: @gblewis1 how does this look

Comment (gblewis1) [reply inbm/Changelog.md]: @tsirlapu maybe something like `- (BREAKING CHANGE) RTC  536910 - [source] Remove ubuntuAptSource INBM configuration tag and underlying code; replaced with source command`

**Review by gblewis1 — APPROVED** [review]

**Comment by nex-maximus** [comment]
<h4>Please provide the PR description rating here on a scale of 1 to 5 (1 being the lowest and 5 being the highest) - 
</h4>

<h3>Summary Report of Code Review Evaluation metrics for the Repository:</h3>

<h4>Code defects Acceptance ratio: 0/0 </h4>

<h4>Code improvements Acceptance ratio: 3/4 (75.0%)</h4>

<h4>Code fixes Acceptance ratio: 0/0 </h4>

| 7 Approvals   | 1 Dispprovals  | 4 Neutral | 1 Response Pending
|----- | --- | --- | --------- |
| https://github.com/intel/intel-inb-manageability/pull/472#discussion_r1467935265<br><br>https://github.com/intel/intel-inb-manageability/pull/472#discussion_r1467935268<br><br>https://github.com/intel/intel-inb-manageability/pull/472#discussion_r1467935270<br><br>https://github.com/intel/intel-inb-manageability/pull/472#discussion_r1473181503<br><br>https://github.com/intel/intel-inb-manageability/pull/472#discussion_r1473183136<br><br>https://github.com/intel/intel-inb-manageability/pull/472#discussion_r1473183404<br><br>https://github.com/intel/intel-inb-manageability/pull/472#discussion_r1474973700 | https://github.com/intel/intel-inb-manageability/pull/472#discussion_r1473718178 | https://github.com/intel/intel-inb-manageability/pull/472#discussion_r1473206273<br><br>https://github.com/intel/intel-inb-manageability/pull/472#discussion_r1473207112<br><br>https://github.com/intel/intel-inb-manageability/pull/472#discussion_r1473228007<br><br>https://github.com/intel/intel-inb-manageability/pull/472#discussion_r1474925542 | https://github.com/intel/intel-inb-manageability/pull/472#discussion_r1467935263 |

**Pull request merged**